src/common/crc32c_ppc.c

```diff
@@ -0,0 +1,105 @@
+#define CRC_TABLE
+#include "include/int_types.h"
+#include "crc32c_ppc_constants.h"
+
+#include <stdlib.h>
+#include <strings.h>
+
+#define VMX_ALIGN	16
+#define VMX_ALIGN_MASK	(VMX_ALIGN-1)
+
+#ifdef REFLECT
+static unsigned int crc32_align(unsigned int crc, unsigned char const *p,
+			       unsigned long len)
+{
+	while (len--)
+		crc = crc_table[(crc ^ *p++) & 0xff] ^ (crc >> 8);
+	return crc;
+}
+#else
+static unsigned int crc32_align(unsigned int crc, unsigned char const *p,
+				unsigned long len)
+{
+	while (len--)
+		crc = crc_table[((crc >> 24) ^ *p++) & 0xff] ^ (crc << 8);
+	return crc;
+}
+#endif
+
+
+#ifdef HAS_POWER8
+unsigned int __crc32_vpmsum(unsigned int crc, unsigned char const *p,
+			    unsigned long len);
+
+/*
+unsigned int crc32_vpmsum(unsigned int crc, unsigned char *p,
+			  unsigned long len)
+*/
+uint32_t ceph_crc32c_ppc_internal(uint32_t crc, unsigned char const *data, unsigned len)
```

Comment:
As you suggested, I changed the name from __crc32_vpmsum to crc32_vpmsum.  

I was not able to mark it static, since that causes compiler errors related to the loader and toc.  The crc32_vpmsum function is called from src/common/crc32c_ppc.c (which becomes crc32c_ppc.o) while the function itself is defined in src/common/crc32c_ppc_asm.S (which becomes crc32c_ppc_asm.o).